File: array/177.py
def numberOfChild(n: int, k: int) -> int:
    no_child = 0
    sec = 0
    forward, backward = True, False

    while sec < k:
        sec += 1

        if forward is True:
            no_child += 1
            if no_child == n - 1:
                forward = False
                backward = True

        elif backward is True:
            no_child -= 1
            if no_child == 0:
                forward = True
                backward = False

    return no_child


print(numberOfChild(28, 49))

File: array/177.cpp
#include <bits/stdc++.h>

using namespace std;

int numberOfChild(int n, int k) {
    int sec = 0;
    int noChild = 0;
    bool forward = true, backward = false;

    while(sec < k) {
        ++sec;

        if(forward) {
            ++noChild;
            if (noChild == n - 1) {
                forward = false;
                backward = true;
            }
        }
        else if(backward){
            --noChild;
            if (noChild == 0) {
                forward = true;
                backward = false;
            }
        }
        
    }

    return noChild;
}


int main() {
    int n = 4, k = 2;
    cout<<numberOfChild(n, k);

    return 0;
}